contribs/gmf/src/dropfile/fileDropZoneComponent.js

```diff
@@ -0,0 +1,166 @@
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import angular from 'angular';
+import {MessageType} from 'ngeo/message/Message.js';
+import './fileAlert.scss';
+
+/**
+ * @type {angular.IModule}
+ * @hidden
+ */
+const module = angular.module('gmffileDropZone', []);
+
+module.run(
+  /**
+   * @ngInject
+   * @param {angular.ITemplateCacheService} $templateCache
+   */
+  ($templateCache) => {
+    // @ts-ignore: webpack
+    $templateCache.put('gmf/dropfile/fileDropZoneComponent', require('./fileDropZoneComponent.html'));
+  }
+);
+
+module.value(
+  'gmfFileDropZoneTemplateUrl',
+  /**
+   * @param {angular.IAttributes} $attrs Attributes.
+   * @return {string} The template url.
+   */
+  ($attrs) => {
+    const templateUrl = $attrs.gmfFileDropZoneTemplateUrl;
+    return templateUrl !== undefined ? templateUrl : 'gmf/dropfile/fileDropZoneComponent';
+  }
+);
+
+/**
+ * @param {angular.IAttributes} $attrs Attributes.
+ * @param {function(angular.IAttributes): string} gmfFileDropZoneTemplateUrl Template function.
+ * @return {string} Template URL.
+ * @ngInject
+ * @private
+ * @hidden
+ */
+function gmfFileDropZoneTemplateUrl($attrs, gmfFileDropZoneTemplateUrl) {
+  return gmfFileDropZoneTemplateUrl($attrs);
+}
+
+/**
+ * Provide file drop zone component.
+ *
+ * Example:
+ *      <gmf-file-drop-zone ng-if="ctrl.dropFileEnabled"></gmf-file-drop-zone>
+ *
+ * If ng-if="ctrl.dropFileEnabled" is used, the component has to be enabled by
+ *       module.value('gmfDropFileEnabled', true);
+ *
+ * @ngdoc component
+ * @ngname gmfFileDropZone
+ */
+const fileDropComponent = {
+  controller: 'gmfFileDropZoneController as ctrl',
+  templateUrl: gmfFileDropZoneTemplateUrl,
+};
+
+module.component('gmfFileDropZone', fileDropComponent);
+
+/**
+ * @param {JQuery} $element Element.
+ * @param {import("ngeo/message/Notification.js").MessageNotification} ngeoNotification Ngeo notification
+ * @param {import("gmf/datasource/ExternalDataSourcesManager.js").ExternalDatSourcesManager} gmfExternalDataSourcesManager The manager of external datasources.
+ * @param {angular.gettext.gettextCatalog} gettextCatalog Gettext catalog.
+ * @constructor
+ * @private
+ * @hidden
+ * @ngInject
+ * @ngdoc controller
+ * @ngname gmfFileDropZoneController
+ */
+function Controller($element, ngeoNotification, gmfExternalDataSourcesManager, gettextCatalog) {
```

Comment:
Using function is the old way to have class in ES.
Can you use `class` to describe this controller ? (see for instance the SearchController in the contribs/gmf/src/search/component.js)